### PR TITLE
Modal view cancelable option

### PR DESCRIPTION
**Pull request by tsonevn**
Modal view cancelable option - Android
Related to https://github.com/NativeScript/NativeScript/issues/6692

### CODE REVIEW

File: tns-core-modules/ui/core/view-base/view-base.d.ts [hunk 1]
```diff
@@ -110,29 +151,47 @@ export abstract class ViewBase extends Observable {
...
     showModal(moduleName: string, context: any, closeCallback: Function, fullscreen?: boolean, animated?: boolean, stretched?: boolean): ViewBase;
 
     /**
+     * @deprecated Use showModal with ShowModalOptions instead.
      * Shows the view passed as parameter as a modal view.
      * @param view - View instance to be shown modally.
      * @param context - Any context you want to pass to the modally shown view. This same context will be available in the arguments of the shownModally event handler.
      * @param closeCallback - A function that will be called when the view is closed. Any arguments provided when calling ShownModallyData.closeCallback will be available here.
      * @param fullscreen - An optional parameter specifying whether to show the modal view in full-screen mode.
+     * @param cancelable - An optional parameter specifying whether the modal view can be dismissed when not in full-screen mode.
```
Comment (vakrilov) [inline tns-core-modules/ui/core/view-base/view-base.d.ts]: Remove this line - there is no `cancelable` param

File: tns-core-modules/ui/core/view/view-common.ts [hunk 2]
```diff
@@ -215,25 +216,42 @@ export abstract class ViewCommon extends ViewBase implements ViewDefinition {
...
+                options = <ShowModalOptions>args[1];
+            } else {
+                // TODO: Add deprecation warning
+                options = {
+                    context: args[1],
+                    closeCallback: args[2],
+                    fullscreen: args[3],
+                    animated: args[4],
+                    stretched: args[5],
+                    cancelable: args[6]
```
Comment (vakrilov) [inline tns-core-modules/ui/core/view/view-common.ts]: No `cancelable` param in the showModal method. You should pass it in `ModalOptions`

**Comment by tsonevn** [comment]
test

**Comment by tsonevn** [comment]
test

**Pull request closed without merging**